dll/win32/shell32/folders/CDesktopFolder.cpp

```diff
@@ -790,15 +837,37 @@ HRESULT WINAPI CDesktopFolder::GetDetailsOf(
         psd->cxChar = DesktopSFHeader[iColumn].cxChar;
         return SHSetStrRet(&psd->str, DesktopSFHeader[iColumn].colnameid);
     }
+    else
+    {   /* iColumn is the listview column number in explorer */
+        /* Swap the ReactOS size and type columns */
+        switch (iColumn)
+        {
+            case 0:                /* name */
+            case 3:                /* modified date */
+            case 4:                /* attributes */
+                break;
+            case 1:                /* size */
+                if (_ILIsSpecialFolder(pidl))
+                    iColumnModified = 5;
+                else
+                    iColumnModified = 2;
+                break;
+            case 2:                /* type */
+                iColumnModified = 1;
+        }
+    }
 
     CComPtr<IShellFolder2> psf;
     HRESULT hr = _GetSFFromPidl(pidl, &psf);
     if (FAILED_UNEXPECTEDLY(hr))
         return hr;
 
-    hr =  psf->GetDetailsOf(pidl, iColumn, psd);
-    if (FAILED_UNEXPECTEDLY(hr))
-        return hr;
+    /* size col and Special Folder */
+    if (DesktopSFHeader[iColumn].colnameid ==  IDS_SHV_COLUMN_SIZE
+        && _ILIsSpecialFolder(pidl))
+        return SHSetStrRet(&psd->str, ""); /* blank col */
```

Comment:
CDesktopFolder is not supposed to know this. `psf->GetDetailsOf` knows the size and any questions about information inside a pidl must come from `psf`.

Reply:
Code now revised with your help. Thanks.